### DISCUSSION TITLE
Tekstfelt feil

### DISCUSSION THREAD

**Discussion by Stakset** [fremtind/jokul#2705] [Annet]
Hvis man klikker på et tekstfelt med feil så er fortsatt den røde fargen der. Burde man gått til active state med feilmelding og ikon fortsatt værende under?

<img width="790" alt="Screenshot 2022-02-04 at 14 53 56" src="https://user-images.githubusercontent.com/72978199/152543106-095fc7c4-46ed-4642-8e8e-4ff9b36f2273.png">

**Comment by wkillerud** [comment]
Jeg heller mot at vi bør beholde bakgrunnsfargen rød fram til feilen er rettet. Jeg har sett på det som en "aktiv"-indikator at feltet får en tykkere ramme rundt seg. Opplever dere at det er vanskelig å følge hvor fokus er?

Jeg finner flere eksempler hvor en visuell feiltilstand ikke fjernes ved focus:
- [Native skjemavalidering i nettleseren](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#validating_against_a_regular_expression)
- [MUI](https://mui.com/components/text-fields/)
- [Fluent UI](https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield)
- [Lightning](https://www.lightningdesignsystem.com/components/input/)

Det sagt – det betyr ikke at _alle_ gjør det, eller at vi må følge samme mønster. Ingen av dem har en bakgrunnsfarge i skjemaet slik som vi har. Åpen for andre synspunkter 😄

**Reply by onordbo** [reply]
Jeg ser ikke på dette som en feil. Fra et interaksjonsståsted må det sees i sammenheng med valideringen helhetlig. En visuell overgang eller indikasjon som viser "godkjent" / akseptert input senker terskelen for at man potensielt skaper nye misforståelser, eller at feil data ender opp på feil sted. **+1 for å beholde rødfargen frem til feilen er rettet av brukeren.**

Når det er sagt, har vi noe visuell respons ved godkjent eller akseptert input? Hva skjer når riktig data står i feltet? Går vi da tilbake til "vanlig" state? Det ligger sikkert noe potensiale i å levle opp brukeropplevelsen med en bekreftende hjelpetekst.

**Reply by wkillerud** [reply]
Det er et eksempel i portalen (via Profil -> Skjemavalidering): https://jokul.fremtind.no/eksempler/skjemavalidering/
too long; didn't click – ja, vi går tilbake til vanlig aktiv state med én gang feltet er gyldig

**Reply by Stakset** [reply]
Jeg mente ikke at det nødvendigvis var feil, selv om overskriften kanskje kan tolkes slik 😅  Mer en undring om opplevelsen ble bedre. Etter å ha sett det eksemplet er jeg enig at man fint kan beholde rødfargen til input er riktig.
